bamboo/bootstrap-tf-deployment.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+set -ex
+. ./bamboo/abort-if-not-pr-or-redeployment.sh
+. ./bamboo/abort-if-skip-integration-tests.sh
+. ./bamboo/abort-if-not-terraform.sh
+
+npm config set unsafe-perm true
+npm install
+. ./bamboo/set-bamboo-env-variables.sh
+
+if [[ $USE_NPM_PACKAGES == true ]]; then
+  echo "***Deploying stack with NPM packages"
+  (cd example && npm install)
+else
+  echo "***Deploying stack with built packages"
+  npm run bootstrap
+fi
+
+echo "Locking stack for deployment $DEPLOYMENT-tf"
+
+cd example
+set +e
+
+# Fetch terraform binary
+if ! curl -o terraform_${TF_VERSION}_linux_amd64.zip https://releases.hashicorp.com/terraform/${TF_VERSION}/terraform_${TF_VERSION}_linux_amd64.zip ; then
+  echo "ERROR: coudn't download terraform script" >&2
+  exit 1
+else
+  unzip -u ./terraform_${TF_VERSION}_linux_amd64.zip
+  chmod a+x ./terraform
+  rm ./terraform_${TF_VERSION}_linux_amd64.zip
+fi
+
+# TODO Necessary in Terraform?
+# Wait for the stack to be available
+node ./scripts/lock-stack.js true $DEPLOYMENT-tf
+LOCK_EXISTS_STATUS=$?
+echo "Locking status $LOCK_EXISTS_STATUS"
```

Comment:
I guess stack locking still applies for common stacks such as master, nightly, from-source etc, glad to hear input.

Reply:
I would assume so - this nicely prevents situations where multiple build plans might be building against significantly different build plans.... however I'm uncertain if that model works as well given how TF handles updates/reversion, but it seems like it should.

Reply:
👍